src/source/to-shift.js

```diff
@@ -1,31 +1,98 @@
 'use strict';
 
-function removeFunctionContents(obj) {
-    obj.type = obj.type.replace(/^(?:Eager|Lazy)/, '');
+function tryRemoveFunctionContents(obj) {
+    let m = obj.type.match(
+        /^(?:Eager|Lazy)(Function(?:Expression|Declaration)|Method|[GS]etter|(ArrowExpression)With(?:FunctionBody|Expression))$/
+    );
 
-    delete obj.isAsync;
-    delete obj.scope;
-    delete obj.contents_skip;
-    delete obj.length;
+    if (!m) {
+        return;
+    }
 
     let { params, body } = obj.contents;
-    delete obj.contents;
 
+    obj.type = m[2] || m[1];
     obj.params = params;
 
-    if (obj.type === 'ArrowExpressionWithExpression') {
-        obj.type = 'ArrowExpression';
-        obj.body = body;
-    } else {
-        if (obj.type === 'ArrowExpressionWithFunctionBody') {
-            obj.type = 'ArrowExpression';
-        }
-        obj.body = {
+    if (m[1] !== 'ArrowExpressionWithExpression') {
+        body = {
             type: 'FunctionBody',
             directives: obj.directives,
             statements: body
         };
     }
+
+    obj.body = body;
 }
 
-function convertObject(key, value) {}
+function convertObject(key, obj) {
```

Comment:
would be nice to document what `this` is, or receive as `parent` and `child` explicitly.

Reply:
given this function itself is not passed to `JSON.parse` (also, the call is in different place than this definition), it's not obvious that this is using `JSON.parse` callback interface.
So, it's better explaining that here.
and single line comment won't harm :)

Reply:
Actually, yes, that would be a good place to write that you're inheriting this signature.